hw/ip/usbdev/data/usbdev.hjson

```diff
@@ -494,6 +502,130 @@
+            of the usbdev peripheral inputs and outputs from software. This
+            is designed to be used for debugging purposes or during chip
+            testing.
+            '''
+      swaccess: "ro",
+      hwaccess: "hwo",
+      hwext: "true",
+      fields: [
+        {
+          bits: "0",
+          name: "rx_dp_i",
+          desc: "USB D+ input."
+        }
+        {
+          bits: "1",
+          name: "rx_dn_i",
+          desc: "USB D- input."
+        }
+        {
+          bits: "2",
+          name: "rx_d_i",
+          desc: "USB Differential input."
+        }
+        {
+          bits: "8",
+          name: "tx_dp_o",
+          desc: "USB D+ output (readback)."
+        }
+        {
+          bits: "9",
+          name: "tx_dn_o",
+          desc: "USB D- output (readback)."
+        }
+        {
+          bits: "10",
+          name: "tx_d_o",
+          desc: "USB Differential output (readback)."
+        }
+        {
+          bits: "11",
+          name: "tx_se0_o",
+          desc: "USB SE0 output (readback)."
+        }
+        {
+          bits: "12",
+          name: "tx_oe_o",
+          desc: "USB OE output (readback)."
+        }
+        {
+          bits: "13",
+          name: "suspend_o",
+          desc: "USB suspend output (readback)."
+        }
+        {
+          bits: "16",
+          name: "pwr_sense",
+          desc: "USB power sense signal."
+        }
+      ]
+    }
+    { name: "phy_pins_drive",
+      desc: '''
+            USB PHY Pin Drive. This register can be used to control the usbdev
+            peripheral inputs and outputs from software. This is designed to be
+            used for debugging purposes or during chip testing.
+            '''
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        {
+          bits: "0",
+          name: "dp_o",
+          desc: "USB D+ output."
+        }
+        {
+          bits: "1",
+          name: "dn_o",
+          desc: "USB D- output."
+        }
+        {
+          bits: "2",
+          name: "d_o",
+          desc: "USB Differential output."
+        }
+        {
+          bits: "3",
+          name: "se0_o",
+          desc: "USB SE0 output."
+        }
+        {
+          bits: "4",
+          name: "oe_o",
+          desc: "USB OE output."
+        }
+        {
+          bits: "5",
+          name: "tx_mode_se_o",
+          desc: "USB TX mode. 0: Differential, 1: Single-ended."
+        }
+        {
+          bits: "6",
+          name: "dp_pullup_en_o",
+          desc: "USB D+ Pullup enable output."
+        }
+        {
+          bits: "7",
+          name: "dn_pullup_en_o",
+          desc: "USB D- Pullup enable output."
+        }
+        {
+          bits: "8",
+          name: "suspend_o",
+          desc: "USB suspend output."
+        }
+        {
+          bits: "16",
+          name: "en",
+          desc: "Enable driving the USB signals."
```

Comment:
I would be a bit more explicit here and say that if the bit is 0, the USB signals are driven by the hardware whereas if the bit is 1, the USB signals are driven by the values specified in this register.

Reply:
done, now it reads:

```
                0: Outputs are controlled by the hardware block.
                1: Outputs are controlled with this register. 
```